eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -822,7 +820,7 @@ boolean renew() {
         EurekaHttpResponse<InstanceInfo> httpResponse;
         try {
             httpResponse = eurekaTransport.registrationClient.sendHeartBeat(instanceInfo.getAppName(), instanceInfo.getId(), instanceInfo, null);
-            logger.debug("{} - Heartbeat status: {}", PREFIX + appPathIdentifier, httpResponse.getStatusCode());
+            logger.debug(PREFIX + "{} - Heartbeat status: {}", appPathIdentifier, httpResponse.getStatusCode());
```

Comment:
All the `PREFIX +` in this class can also be moved to `PREFIX + appPathIdentifier`

Reply:
In this case, because `PREFIX` is a constant variable (as defined in the Java Language Specification, section 4.12.4 `final` Variables), having `PREFIX` concatenated in the format string avoids a runtime string concatenation as the compiler will compute the constant expression and treat the same way as a literal; from section 3.10.5 String Literals:

> Strings computed by constant expressions (§15.28) are computed at compile time and then treated as if they were literals.